scripts/perftune.py

```diff
@@ -643,6 +643,77 @@ def __intel_irq_to_queue_idx(self, irq):
         else:
             return sys.maxsize
 
+    def __mlx_irq_to_queue_idx(self, irq):
+        """
+        Return the HW queue index for a given IRQ for Mellanox NICs in order to sort the IRQs' list by this index.
+
+        Mellanox NICs have the IRQ which name looks like
+        this:
+        mlx5_comp23
+             <bla-bla>mlxX_comp<index>
+        or this:
+        mlx4-6
+             <bla-bla>mlxX-<index>
+
+        :param irq: IRQ number
+        :return: HW queue index for Mellanox NICs and 0 for all other NICs
+        """
+        mlx_fp_irq_re = re.compile("mlx\d_comp(\d+)|mlx\d\-(\d+)")
+
+        m = mlx_fp_irq_re.search(self.__irqs2procline[irq])
+        queue=m.group(2)
+        return int(queue)
+
+    def __get_driver_name(self, iface):
+        """
+        :param iface: Interface to check
+        :return: driver name from ethtool
+        """
+
+        driver_name = ''
+        ethtool_i_lines = run_read_only_command(['ethtool', '-i', iface]).splitlines()
+        driver_re = re.compile("driver:")
+        driver_lines = list(filter(lambda one_line: driver_re.search(one_line), ethtool_i_lines))
+
+        if driver_lines:
+            if len(driver_lines) > 1:
+                raise Exception("More than one 'driver:' entries in the 'ethtool -i {}' output. Unable to continue.".format(iface))
+
+            driver_name = driver_lines[0].split()[1].strip()
+
+        return driver_name
+
+    def __set_driver_rx_queues(self, iface, queue_count):
+        """
+        :param iface: interface to set
+        :param queue_count: number of RX queues to set
+        """
+        try:
+                perftune_print("Limiting RX queues on {} ...".format(iface))
+                run_one_command(['ethtool','-L', iface, 'rx', str(queue_count)], stderr=subprocess.DEVNULL)
```

Comment:
Sometimes you need to set `combined` in order to set `rx` queues count. - Many vendors don't allow configuring different Rx and Tx queues number. Therefore they use `combined` value.

Look in the `man ethtool`

Reply:
ok, so that's why this is called __set_driver_rx_queues
later in the code it's only used with mlx driver prefix only (which support setting only rx, which is what we want)

do I need to add a check here for ethtool if it has combined or tx rx ?
(I think -l will just show channels, so I can detect if it supports rx and use rx, or do nothing)

Reply:
The driver can support either `rx + tx` or `combined`. In the later case setting "rx queues number" means setting the `combined` value. 

You can choose to implement setting `rx`, `tx` and `combined` in separate functions and then call them depending on the driver "mode" which you can detect.

Regardless, the current implementation (covering only non-combined case) is partial is not enough.

BTW, I do recall that one of mlx4 or mlx5 supports the former mode and another - the later.
So, even in the context of Mellanox cards you need to support both.